components/cam/src/dynamics/se/inital.F90

```diff
@@ -32,7 +37,18 @@ subroutine cam_initial(dyn_in, dyn_out, NLFileName)
 
    ! Define physics data structures
    if(par%masterproc  ) write(iulog,*) 'Running phys_grid_init()'
-   call phys_grid_init( )
+   call phys_grid_init()
+
+   ! Initialize index values for advected and non-advected tracers
+   call phys_register()
+
+   ! Frontogenesis indices
+   if (use_gw_front) then
```

Comment:
Why not push these pbuf_add_field() calls down into phys_register() since there are already a lot of pbuf_add_field() calls in there?

Reply:
I think that I asked someone about this, and was encouraged to leave these pbuf_add_field calls in the dynamics. In any case, the relevant variables (frontgf_idx and frontga_idx) are defined in dyn_comp.F90, and are not in all of the dycores. While we are not supporting any dycore but SE, I am trying not to deliberately introduce anything  that would prevent restoring the functionality of one of these other dycores. There was less of a logicial structure change by doing it this way. If one of the EAM developers wants to make this move, I don't have an opinion (beyond what I just said).
In summary - I did not think that it was my place to propose such a change, and it was not necessary for what I needed.

Reply:
ok, right, this is obviously outside the scope of this PR.

Reply:
FYI - I asked this identical question in github issue  #3538 , and you were the only one to respond :-). After seeing that not all dycores support the  frontal gw scheme, I gave up looking at adding it to phys_register.

Reply:
Sounds like we should just take a machete to these other dycor options...